src/java/org/apache/cassandra/db/marshal/VectorType.java

```diff
@@ -365,6 +366,7 @@ public abstract class VectorSerializer extends TypeSerializer<List<T>>
 
         public abstract <V> List<V> split(V buffer, ValueAccessor<V> accessor);
         public abstract <V> V serializeRaw(List<V> elements, ValueAccessor<V> accessor);
+        public abstract float[] deserializeFloatArray(ByteBuffer input);
```

Comment:
Does this method duplicate `composeAsFloat`?

Reply:
No, `composeAsFloat` composes a `float[]` for the specific `VectorType`. As in, it uses the `dimension` specified for the `VectorType` instance this is called on. The `deserializeFloatArray` method is not specific to single instance and gets the dimension from the buffer passed to is that was previously generated by a call to `serialize`.

Reply:
I see, but `deserializeFloatArray ` is an instance method with access to dimension. It produces the same output for the same input as `composeAsFloat` unless the input is of a different dimension than the type instance it belongs to, I think. Can this ever happen, that we call it with an input of different size? If that's the case, it should probably be a static method. If not, then one of the implementations seems redundant.

Reply:
Having looked again, I see that the `deserializeFloatArray` is only implemented for the `Fixed` variant of the serializer, so it can be removed in favour of the `composeAsFloat`. The `deserializeFloatArray` was added because it could shortcut the deserialization by using the `FloatBuffer` to get the array. I have added a `ValueAccessor.toFloatArray` method in order to support this optimisation for `ByteBuffer`s.